src/Inventory.ts

```diff
@@ -124,75 +116,95 @@ export function retrieveTools (bot: Bot, options: ToolRetrievalOptions, cb: Call
  * @param bot - The bot to move.
  * @param location - The location to move to.
  * @param cb - The callback to run when finished.
+ * @returns {Promise<void>}
  */
-function gotoChest (bot: Bot, location: Vec3, cb: Callback): void {
-  // @ts-expect-error
+async function gotoChest (bot: Bot, location: Vec3, cb?: Callback): Promise<void> {
   const pathfinder = bot.pathfinder
+  return await new Promise((resolve, reject) => {
+    pathfinder.setGoal(new goals.GoalGetToBlock(location.x, location.y, location.z))
 
-  pathfinder.setGoal(new goals.GoalBlock(location.x, location.y, location.z))
+    const events = new TemporarySubscriber(bot)
+    events.subscribeTo('goal_reached', () => {
+      events.cleanup()
+      resolve()
+      if ((cb != null) && typeof cb === 'function') cb()
+    })
 
-  const events = new TemporarySubscriber(bot)
-  events.subscribeTo('goal_reached', () => {
-    events.cleanup()
-    cb()
-  })
+    events.subscribeTo('path_update', (results: ComputedPath) => {
+      if (results.status === 'noPath') {
+        events.cleanup()
+        const err = error('NoPath', 'No path to target block!')
+        reject(err)
+        if ((cb != null) && typeof cb === 'function') cb(err)
+      }
+    })
 
-  events.subscribeTo('path_update', (results: ComputedPath) => {
-    if (results.status === 'noPath') {
+    events.subscribeTo('goal_updated', () => {
       events.cleanup()
-      cb(error('NoPath', 'No path to target block!'))
-    }
-  })
-
-  events.subscribeTo('goal_updated', () => {
-    events.cleanup()
-    cb(error('PathfindingInterrupted', 'Pathfinding interrupted before item could be reached.'))
+      const err = error('PathfindingInterrupted', 'Pathfinding interrupted before item could be reached.')
+      reject(err)
+      if ((cb != null) && typeof cb === 'function') cb(err)
+    })
   })
 }
 
-function pullFromChest (bot: Bot, chestPos: Vec3, options: ToolRetrievalOptions, cb: (err: Error | undefined, gotItem: boolean) => void): void {
-  const chestBlock = bot.blockAt(chestPos)
-  if (chestBlock == null) {
-    cb(error('UnloadedChunk', 'Chest is located in an unloaded chunk!'), false)
-    return
-  }
-
-  const chest = bot.openChest(chestBlock)
-
-  chest.once('open', () => {
-    let itemsToPull: Item[] = []
-    for (const item of chest.items()) {
-      if (options.toolFilter(item)) { itemsToPull.push(item) }
-    }
-
-    if (itemsToPull.length === 0) {
-      cb(undefined, false)
+/**
+ * Pull items from a given chest
+ * @param bot Bot
+ * @param chestPos Chest Position
+ * @param options Tool Retrieval Options {@link ToolRetrievalOptions}
+ * @param cb Optional callback
+ * @returns {Promise<boolean>}
+ */
+async function pullFromChest (bot: Bot, chestPos: Vec3, options: ToolRetrievalOptions, cb?: (err: Error | undefined, gotItem: boolean) => void): Promise<boolean> {
+  return await new Promise(async (resolve, reject) => {
```

Comment:
you shouldn't ever need to use async with promises, at that point get rid of the promise and use an async function, and you are already in an async function, so maybe just get rid of the promise and use the async function

Reply:
getting rid of this will fix both lint errors